drivers/sensors/paw3222.c

```diff
@@ -0,0 +1,177 @@
+/* Copyright 2024 Colin Lam (Ploopy Corporation)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "paw3222.h"
+#include "wait.h"
+#include "debug.h"
+#include "gpio.h"
+#include "math.h"
+#include "spi_master.h"
+#include <print.h>
+#include "pointing_device_internal.h"
+
+#define MSB1 0x80
+#define MSB0 0x7F
+
+const pointing_device_driver_t paw3222_pointing_device_driver = {
+    .init       = paw3222_init,
+    .get_report = paw3222_get_report,
+    .set_cpi    = paw3222_set_cpi,
+    .get_cpi    = paw3222_get_cpi,
+};
+
+void paw3222_write(uint8_t reg_addr, uint8_t data) {
+    spi_start(PAW3222_CS_PIN, false, 3, PAW3222_SPI_DIVISOR);
+    wait_us(1); // Tncs_lead
+    spi_write(reg_addr | MSB1);
+    spi_write(data);
+    wait_us(1); // Tncs_lag
+    spi_stop();
+    // wait_us(10);
```

Comment:
```suggestion
```

Reply:
Done in [ff65fd0](https://github.com/qmk/qmk_firmware/pull/25763/commits/ff65fd085f10b82f1a17e161c5da6f06e4f0b39b).